Updating section to support setting the new right variant.

jacobb84 opened the pull request:
Adding red boarder when image component is added.

Like before, this depends on the design system update being merged first. The read boarder shows automatically when an image component is added.

jacobb84 commented:
Hi Aaron, concerning the property change, I did consider that but I ran I scan of the behavior_settings field in the database backup we had and the only place it was being used was the home page. I thought about doing an update hook, but I wasn't sure if the risk of altering something it shouldn't was worth it vs just updating the one property post deployment. Or maybe we do an update hook and just target the home page?

For the ask, it's based on this part of the figma design: https://www.figma.com/design/pfhQCvR2YmyHyKjEXS9gnK/23-24_ILR-WIP?node-id=8937-643&m=dev

It was requested we update the card to support inverting the header / content part of the card and also add the red border around the card when an image widget is added to it.

jacobb84 commented:
Ah drat, ok my local DB must have been more out of date then I realized. In that case we'd definitely want to do an update hook

a-fro commented:
@jacobb84 I'm not sure we will need an update hook, because I'm not clear how these code changes relate to the new card variant pictured above. I'd think that using the new variant, which an editor would choose in the "Card settings" behavior, would give the design variation above. I'm not clear on how it would relate to the section settings for heading alignment. 

If you agree with this approach, I wonder if you have suggestions what to call it. Naming is hard. Side by side?

jacobb84 commented:
@a-fro Let me back a bit. We originally had a call with Dave and he had two changes he wanted to this Side By Side widget. One was the red border, which is what the figma link points to. And you're right that this setting has nothing to do with that.  The second change he wanted, was the ability to support this layout:

<img width="1819" height="756" alt="image" src="https://github.com/user-attachments/assets/bb3f5b3b-e7bc-45f6-bf03-eb195316eef5" />

For the red border, I have it applying that style if an image component is added to that Section paragraph instance. I can absolutely make it into a toggle for the content author if it shouldn't have that red border all the time. I can also make it into a variant for the image component in the design pattern if we go that route. 

For the second change, I added the markup to support the right alignment in this PR. https://github.com/ilrWebServices/union/pull/41. When I looked at the site, I saw that the setting that was left aligning the header, was being controlled by section settings

<img width="1177" height="453" alt="image" src="https://github.com/user-attachments/assets/3e066701-2152-4eb2-ab68-0880fd12825b" />

I considered just adding a "Place heading on right" checkbox, but I thought that might be confusing to the content author, since they'd have to remember to specifically uncheck the other checkbox when they made changes. So I removed the heading_left setting and replaced it with the header_alignment setting, and added left and right options so they could use both. To your point, I didn't realize how many spots already had used the header_left setting, so we'd need the update hook to avoid breaking things.

Does this make more sense? Maybe it would be helpful if I broke these into separate PRs?

a-fro commented:
Thanks for walking me through your thinking. This is super helpful. I'd love to have a short conversation to discuss our mental model and make an implementation plan together. HMU on Slack and hopefully we can find a time.

jeffam commented:
Closed in favor of https://github.com/ilrWebServices/ilr/pull/50
